### PR TITLE
Do not show "Routing options enabled"  when using ruler

### LINKED ISSUE #6579: Don't show "Routing options enabled" for Ruler routes
**Steps to reproduce:**
1. Open settings, Routing options and enable some road avoidance settings
2. Build a route using Ruler route type

**Actual result:**

"Routing options enabled" panel is shown

Screenshot:

**Expected result:**
Panel should not be shown.

### PR DESCRIPTION
**Pull request by dmfrodrigues**
Do not show "Routing options enabled"  when using ruler.

Based on https://github.com/organicmaps/organicmaps/pull/3601.

Closes #6579.

### CODE REVIEW

**Review by Jean-BaptisteC — APPROVED** [review]

File: android/app/src/main/java/app/organicmaps/routing/RoutingController.java [hunk 1]
```diff
@@ -198,7 +198,7 @@ private void processRoutingEvent()
 
   private boolean isDrivingOptionsBuildError()
   {
-    return !ResultCodesHelper.isMoreMapsNeeded(mLastResultCode) && RoutingOptions.hasAnyOptions();
+    return !ResultCodesHelper.isMoreMapsNeeded(mLastResultCode) && RoutingOptions.hasAnyOptions() && !isRulerRouterType();
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/routing/RoutingController.java:201]: `isDrivingOptionsBuildError` looks confusing for the given issue, is it the right place to insert the check? When is `isDrivingOptionsBuildError` called? Maybe it's better to rename it? Or add the `isRulerRouterType` check outside of this function call?

Should a similar line in `android/app/src/main/java/app/organicmaps/routing/RoutingPlanController.java` also be updated?

Comment (dmfrodrigues) [reply android/app/src/main/java/app/organicmaps/routing/RoutingController.java:201]: I also noticed the function name is a bit odd, but figured it should be the right place based on the fact a router filter was previously present here: https://github.com/organicmaps/organicmaps/pull/3601. I'd also agree to change its name to something more obvious; any suggestions?

As for calling in `RoutingPlanController.java`, I figured I could not reproduce a case where the warning showed, so I didn't change anything else. Will change.

**Review by vng — APPROVED** [review]

**Comment by dmfrodrigues** [comment]
@biodranik can you take another look into this PR?

**Review by biodranik — APPROVED** [review]
I'm very sorry for the delayed review. Thank you very much for your contribution! Welcome to the OM!

**Pull request merged**